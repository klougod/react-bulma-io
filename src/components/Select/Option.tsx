import React, { ReactNode, OptionHTMLAttributes } from 'react'

interface OptionProps extends OptionHTMLAttributes<HTMLOptionElement> {
  text?: ReactNode
}

export const Option = ({ text, ...rest }: OptionProps) => (
  <option {...rest}>{text}</option>
)

Option.defaultProps = {
  text: '',
  value: '',
  className: '',
  disabled: false
}

export default Option
