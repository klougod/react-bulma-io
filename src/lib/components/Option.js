import React from 'react'
import PropTypes from 'prop-types'

export const Option = ({ value, text, className }) => (
  <option className={className} value={value}>
    {text}
  </option>
)

Option.defaultProps = {
  text: '',
  value: '',
  className: ''
}

Option.propTypes = {
  text: PropTypes.string,
  value: PropTypes.any,
  className: PropTypes.string
}

export default Option