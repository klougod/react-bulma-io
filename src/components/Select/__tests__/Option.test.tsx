import React from 'react'
import Enzyme from 'enzyme'

import Option from '../Option'

describe('Option component test', function () {
  it('should render without crash with option', function () {
    const optionMock = Enzyme.shallow(<Option />)
    expect(optionMock).toMatchSnapshot()
  })

  it('should render with props', function () {
    const optionMockProps = {
      text: 'option text',
      value: 'option value',
      className: 'optionClass'
    }
    const optionMock = Enzyme.shallow(<Option {...optionMockProps} />)
    expect(optionMock).toMatchSnapshot()
  })

  it('should render disabled', function () {
    const optionMockProps = {
      disabled: true
    }
    const optionMock = Enzyme.shallow(<Option {...optionMockProps} />)
    expect(optionMock).toMatchSnapshot()
  })
})
