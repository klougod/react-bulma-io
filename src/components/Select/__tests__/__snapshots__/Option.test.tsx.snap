// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component test should render disabled 1`] = `
<option
  className=""
  disabled={true}
  value=""
/>
`;

exports[`Option component test should render with props 1`] = `
<option
  className="optionClass"
  disabled={false}
  value="option value"
>
  option text
</option>
`;

exports[`Option component test should render without crash with option 1`] = `
<option
  className=""
  disabled={false}
  value=""
/>
`;
